code/src/components/ScanBarcode.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react'
+import { BarcodeScanner } from 'components/BarcodeScanner'
+import { useDispatch } from 'react-redux'
+import { getProduct } from 'reducers/productsReducer'
+import styled from 'styled-components';
+import { Product } from 'components/Product'
+
+export const ScanBarcode = () => {
+  const [showScanner, setShowScanner] = useState(false)
+  const dispatch = useDispatch()
+
+  return (
+    <Section>
+      <Wrapper>
+        {!showScanner && (
+          <Ean src="../assets/ean.jpg" alt="ean" />
+        )}
+        {showScanner && (
+          <BarcodeScanner onDetected={(code) => {
+            setShowScanner(false)
+            dispatch(getProduct(code))
+          }} />
+        )}
+        <Product />
```

Comment:
The product info overlaps with the button right now, maybe just mount `<Product/>` in App.js below `<ScanBarcode/>` instead?